pkg/controllers/status/cluster_status_controller.go

```diff
@@ -239,15 +246,19 @@ func (c *ClusterStatusController) updateStatusIfNeeded(cluster *clusterv1alpha1.
 
 // buildInformerForCluster builds informer manager for cluster if it doesn't exist, then constructs informers for node
 // and pod and start it. If the informer manager exist, return it.
-func (c *ClusterStatusController) buildInformerForCluster(cluster *clusterv1alpha1.Cluster) (genericmanager.SingleClusterInformerManager, error) {
-	singleClusterInformerManager := c.InformerManager.GetSingleClusterManager(cluster.Name)
-	if singleClusterInformerManager == nil {
-		clusterClient, err := c.ClusterDynamicClientSetFunc(cluster.Name, c.Client)
+func (c *ClusterStatusController) buildInformerForCluster(clusterClient *util.ClusterClient) (typedManager.SingleClusterInformerManager, error) {
+	if c.GenericInformerManager.GetSingleClusterManager(clusterClient.ClusterName) == nil {
```

Comment:
// cluster-status-controller will initialize the generic informer manager
// mainly because when the member cluster is joined, the dynamic informer required by the member cluster 
// to distribute resources is prepared in advance
// in that case execution-controller can distribute resources without waiting.

Reply:
done